### ISSUE TITLE
Am I doing an anti-pattern ?

### ISSUE THREAD

**Issue by LouisCuvelier** [tailwindcss/discuss#345]
Hello,

I feel like I'm doing an anti-pattern thing with this kind of CSS.

```css
.font-title-core {
  @apply font-sans leading-none text-neutral-900;
}

.font-title-0 {
  @apply font-title-core text-5xl;

  @screen sm {
    & {
      @apply text-6xl;
    }
  }
}

.font-title-1 {
  @apply font-title-core text-4xl;

  @screen sm {
    & {
      @apply text-5xl;
    }
  }
}
```

I tried to make this code as a Tailwind plugin but I had this error : `X is included in multiple rulesets`. I searched and I found [this issue](https://github.com/tailwindcss/discuss/issues/155).

So, stacking CSS like this with @apply is a bad thing right ?

**Comment by hacknug** [comment]
I've done this multiple times and haven't encountered any issues with it. Just make sure everything related to each class is defined inside the same block and you should be good.

Also, not sure what postcss plugins you're using but there's probably no need for you to have the `& {}` block in there. Simply write your `@apply` inside `@screen` if you want:

```pcss
.font-title-core {
  @apply font-sans leading-none text-neutral-900;
}

.font-title-0 {
  @apply font-title-core text-5xl;
  @screen sm { @apply text-6xl; }
}

.font-title-1 {
  @apply font-title-core text-4xl;
  @screen sm { @apply text-5xl; }
}
```

**Comment by LouisCuvelier** [comment]
Thanks for your answer !

Yup, you're right. I didn't pay attention

**Comment by LouisCuvelier** [comment]
@hacknug I thought about it again and I found that stacking with `@apply` isn't a good solution.

**Why ?** 
If I reuse my exemples, I have 7 different `font-title-X` (from 0 to 6). So, I'm coping each properties from `font-title-core` inside each `font-title-X`. It is causing a lot of duplication and unnecessary CSS. So, a heavier CSS file.

To avoid this, we shouldn't stacking @apply and use it, instead, in HTML like this : 
```html
<h1 class="font-title-core font-title-X">Hello</h1>
```
This way, we are really respecting "utility-first".

@adamwathan, I'm curious about your point of view.

**Comment by hacknug** [comment]
Of course, but sometimes you need to style things and you don’t have access to the markup. In those scenarios having helpers like those helps a lot imho. Then purgecss takes care of removing things you don’t need for you (although you’ll have to set it up).

**Comment by adamwathan** [comment]
@LouisCuvelier I totally agree the multi-class approach is the way to go whenever possible 👍 Not so much because of filesize concerns but moreso just because I believe composition belongs in the DOM generally.

Here's one benefit of the multi-class approach in your case... Say for one reason or another, you needed to target all of the `font-title-X` classes inside some element and override the color.

When stacking apply, you'd end up having to do something like this:

```css
.parent .font-title-1 { @apply text-white }
/* ... */
.parent .font-title-15 { @apply text-white }
```

Using the multi-class approach, you only need to write a single selector, because each instance of `font-title-core` has a shared selector:

```css
.parent .font-title-core { @apply text-white }
```

Good relevant read here: http://nicolasgallagher.com/about-html-semantics-front-end-architecture/#component-modifiers

**Comment by LouisCuvelier** [comment]
@adamwathan @hacknug Thank you for your points of view. I appreciate a lot. 🙏🏼
